src/Features/Core/Portable/Completion/CompletionChange.cs

```diff
@@ -44,33 +58,55 @@ private CompletionChange(ImmutableArray<TextChange> textChanges, int? newPositio
         /// <param name="includesCommitCharacter">True if the changes include the typed character that caused the <see cref="CompletionItem"/> to be committed.
         /// If false, the completion host will determine if and where the commit character is inserted into the document.</param>
         /// <returns></returns>
-        public static CompletionChange Create(ImmutableArray<TextChange> textChanges, int? newPosition = null, bool includesCommitCharacter = false)
+        [Obsolete("Use Create overload that only takes a single TextChange")]
+        public static CompletionChange Create(
+            ImmutableArray<TextChange> textChanges,
+            int? newPosition = null,
+            bool includesCommitCharacter = false)
         {
             return new CompletionChange(textChanges, newPosition, includesCommitCharacter);
         }
 
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+        public static CompletionChange Create(
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+#pragma warning restore RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
+            TextChange textChange,
+            int? newPosition = null,
+            bool includesCommitCharacter = false)
+        {
```

Comment:
There is no source or binary compat concern with this overload being added.  The first parameter is non-optional in each overload.  And the first parameter will always distinguish exactly which overlaod is being picked.  The RSXXXX warning are for properly adding overloads where the initial parameters share the same types.